pkg/sql/logictest/testdata/logic_test/notice

```diff
@@ -23,3 +23,13 @@ SELECT crdb_internal.notice('debug1', 'now you see this'), crdb_internal.notice(
 ----
 DEBUG1: now you see this
 WARNING: and you see this
+
+statement ok
+CREATE DATABASE d;
+CREATE TABLE d.t (x int)
+
+query T noticetrace
+ALTER TABLE d.t RENAME TO d.t2
+----
+NOTICE: renaming tables with a qualification is deprecated
+HINT: use ALTER TABLE d.public.t RENAME TO t2 instead
```

Comment:
ideally we re-use the same syntax (e.g. d.t in this case)

Reply:
I thought it would be annoying to do this because of how we modify table names in place in AST's but it was actually fine